Architecture.md

```diff
@@ -24,17 +24,18 @@ To summarize, there are six main parts to the PyO3 codebase.
    - [`src/instance.rs`] and [`src/types`]
 3. [`PyClass` and related functionalities.](#3-pyclass-and-related-functionalities)
    - [`src/pycell.rs`], [`src/pyclass.rs`], and more
-4. [Protocol methods like `__getitem__`.](#4-protocol-methods)
-   - [`src/class`]
-5. [Procedural macros to simplify usage for users.](#5-procedural-macros-to-simplify-usage-for-users)
-   - [`src/derive_utils.rs`], [`pyo3-macros`] and [`pyo3-macros-backend`]
-6. [`build.rs` and `pyo3-build-config`](#6-buildrs-and-pyo3-build-config)
+4. [Procedural macros to simplify usage for users.](#5-procedural-macros-to-simplify-usage-for-users)
```

Comment:
Shouldn't the number in the link anchor change as well?

Reply:
Good catch, fixed 👍